tangelo/linq/translator/translate_qubitop.py

```diff
@@ -50,10 +62,15 @@ def translate_operator(qubit_operator, source, target, n_qubits=None):
         if source not in TO_TANGELO:
             raise NotImplementedError(f"Qubit operator conversion from {source} to {target} is not supported.")
         qubit_operator = TO_TANGELO[source](qubit_operator)
+        # Clean the very small terms.
+        qubit_operator.compress()
     if target != "tangelo":
         if target not in FROM_TANGELO:
             raise NotImplementedError(f"Qubit operator conversion from {source} to {target} is not supported.")
         n_qubits = count_qubits(qubit_operator) if n_qubits is None else n_qubits
-        qubit_operator = FROM_TANGELO[target](qubit_operator, n_qubits)
+        try:
```

Comment:
Would you prefer adding `n_qubits` to all underlying functions instead of putting a try block here?

Reply:
I'd rather have a support for `n_qubits` everywhere possible (if we don't know how to enforce it for a specific backend, we can raise a warning if the user provides a value).

Let's make sure `n_qubits` is tagged as optional when it can be.

Reply:
Alright, I'll try to leverage the `n_qubits` in the function to explicitly set the number of qubits. However I will keep it optional.